tests/test_db_runtime_security.py

```
@@ -144,914 +174,955 @@ def _connect_side_effect(*args, **kwargs):
+        "query-only",  # Invalid character
+        "recursive triggers",  # Space in name
+        "123invalid",  # Starts with number
+        "",  # Empty string
+    ],
+)
+def test_pragma_validation_invalid_names(pragma, db_path):
+    """Test that invalid pragma names are rejected."""
+    with pytest.raises(ValueError) as cm:
+        manager = DatabaseManager(db_path, extra_pragmas={pragma: "value"})
+        with manager.read() as cursor:
+            cursor.execute(TEST_SQL_SELECT_ONE)
```

Ruff (0.15.12)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)